### ISSUE TITLE
EmptyConditionalBody autocorrect can result in breaking logic changes

### ISSUE THREAD

**Issue by CvX** [rubocop/rubocop#12669]
Looking at EmptyConditionalBody spec, some autocorrect scenarios (in `expect_correction` blocks) actually result in logic-breaking changes. For example:

https://github.com/rubocop/rubocop/blob/3f24b822a37f08e54a3e076e3a091f808adc6fa9/spec/rubocop/cop/lint/empty_conditional_body_spec.rb#L115-L133

To make it clearer here's the same case but with more real-world-y variable names:

```rb
# before
if show_version
  puts VERSION
elsif dry_run
elsif delete_files
  `rm -rf *`
end

# after
if show_version
  puts VERSION
elsif delete_files
  `rm -rf *`
end
```

After an autocorrect run, `dry_run` flag is no longer used which is a breaking change.

## RuboCop version

`1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.2.1) [arm64-darwin22]`
/the latest on the main branch

**Comment by vlad-pisanov** [comment]
The autocorrect is already marked as unsafe, but it could be improved to replace internal `elsif` branch bodies with `nil`, as opposed to removing them.

```ruby
if show_version
  puts VERSION
elsif dry_run # internal branch, replace with `nil`
  nil
elsif delete_files
  `rm -rf *`
end
```
